tasks/winbuildscripts/Generate-Pyzip.ps1

```diff
@@ -10,49 +10,75 @@ param (
 # https://www.python.org/ftp/python/3.8.0/python-3.8.0-amd64.exe
 # https://www.python.org/ftp/python/3.8.0/python-3.8.0.exe
 
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
 $maj, $min, $patch = $Version.Split(".")
 Write-Host -ForegroundColor Green "Installing Python version $maj $min $patch"
 
-if($maj -eq "2") {
-    $url = "https://www.python.org/ftp/python/$Version/python-$Version.amd64.msi"
-} elseif ($maj -eq "3"){
-    $url = "https://www.python.org/ftp/python/$Version/python-$Version-amd64.exe"
-} else {
-    Write-Host -ForegroundColor Red "Unknown major version $Maj.  I don't know how to do this"
-    exit 1
-}
-$outzip = "python-windows-$Version-amd64.zip"
-if($x86 -eq $true){
-    if($maj -eq "2") {
-        $url = "https://www.python.org/ftp/python/$Version/python-$Version.msi"
+function Get-Installer-Url {
+    param (
+        [string]$version
+    )
+
+    $maj, $min, $patch = $Version.Split(".")
+    if ($maj -eq "2") {
+        $url = "https://www.python.org/ftp/python/$Version/python-$Version.amd64.msi"
     } elseif ($maj -eq "3"){
-        $url = "https://www.python.org/ftp/python/$Version/python-$Version.exe"
+        $url = "https://www.python.org/ftp/python/$Version/python-$Version-amd64.exe"
+    } else {
+        Write-Host -ForegroundColor Red "Unknown major version $Maj.  I don't know how to do this"
+        exit 1
     }
+    
+    if ($x86 -eq $true) {
+        if ($maj -eq "2") {
+            $url = "https://www.python.org/ftp/python/$Version/python-$Version.msi"
+        } elseif ($maj -eq "3"){
```

Comment:
Linting nit: the spacing between conditions and opening brackets is inconsistent (sometimes there's a space, eg. lines 33 and 34, sometimes there isn't, like on this line). Might be worth passing the file through a linter at some point in the future.